pkg/model/openstackmodel/servergroup.go

```diff
@@ -191,39 +195,21 @@ func (b *ServerGroupModelBuilder) buildInstances(c *fi.CloudupModelBuilderContex
 		c.AddTask(instanceTask)
 
 		// Associate a floating IP to the instances if we have external network in router
-		// and respective topology is "public"
+		// and respective subnet is "Public" or "Utility".
 		if b.Cluster.Spec.CloudProvider.Openstack.Router != nil {
 			if ig.Spec.AssociatePublicIP != nil && !fi.ValueOf(ig.Spec.AssociatePublicIP) {
 				continue
 			}
-			switch ig.Spec.Role {
-			case kops.InstanceGroupRoleBastion:
+			if havePublicSubnet {
```

Comment:
the only problem with this setup is that now if people had "Private" topology subnets (and bastion in those) it means that after this PR kops will create bastion to private subnet without floating ip and there is no way to connect to it. Openstack provider does not create loadbalancer just for bastion like AWS (as I see it its waste of resources). I would still create floatingip always for bastion to have backwards compatibility.

Reply:
the real problem here is that openstack is not similar for each company, sometimes its possible that people can connect already to fixed ips, sometimes not (like in our case). The network setup varies a lot between companies.

If we check old code: it created floatingip always for bastion, but only for master/nodes if those had Public topology.

Reply:
"Utility" subnets are not either used in openstack, at least we have not used those in anywhere

Reply:
Thank you for the feedback and context. I have changed the PR to always create FloatingIP for bastions.

Would it be appropriate to change the subnet types in the failing tests? Do you think there might be users with private topology but subnets incorrectly marked as type public?

Reply:
its little bit difficult how people are thinking what is the different between private and public topologies in OpenStack. The problem is that OpenStack can be installed in so many different ways.

For me and how I am thinking it should be:

private topology:
- instances does have ip address from internal network

public topology:
- instances does have ip address from internal network
- instances do have also floating ip address (from external network)

However, in case of private topology in some OpenStack installations it is already possible to connect to instances. It depends how the network is installed.